packages/shared/src/components/filters/ContentTypesFilter.tsx

```diff
@@ -1,40 +1,172 @@
 import React, { ReactElement } from 'react';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 import { FilterCheckbox } from '../fields/FilterCheckbox';
 import { useAdvancedSettings } from '../../hooks/feed';
 import useFeedSettings from '../../hooks/useFeedSettings';
 import { getVideoSetting } from './helpers';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../typography/Typography';
+import { Divider, SharedFeedPage } from '../utilities';
+import { LanguageDropdown } from '../profile/LanguageDropdown';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { ContentLanguage } from '../../lib/user';
+import { UPDATE_USER_PROFILE_MUTATION } from '../../graphql/users';
+import { gqlClient } from '../../graphql/common';
+import { useToastNotification } from '../../hooks';
+import { labels } from '../../lib';
+import { useFeature } from '../GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { OtherFeedPage, RequestKey } from '../../lib/query';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent, TargetType } from '../../lib/log';
 
 export function ContentTypesFilter(): ReactElement {
+  const postTitleLanguageFeature = useFeature(feature.postTitleLanguage);
+  const queryClient = useQueryClient();
   const { advancedSettings, isLoading } = useFeedSettings();
   const { selectedSettings, onToggleSettings } = useAdvancedSettings();
+  const { user, updateUser } = useAuthContext();
+  const { displayToast } = useToastNotification();
+  const { logEvent } = useLogContext();
 
   const videoSetting = getVideoSetting(advancedSettings);
 
+  const { mutate: onLanguageChange } = useMutation(
```

Comment:
since it's quite a bloat would consider it in a hook,. but not highly important

Reply:
Moved here 29b6d9fbf09b1fa3aa9bb4eff2a714e7f0700b81